alvr/session/src/settings.rs

```diff
@@ -1165,6 +1184,9 @@ pub fn session_settings_default() -> SettingsDefault {
                     variant: RateControlModeDefaultVariant::Cbr,
                 },
                 filler_data: false,
+                h264_profile: H264ProfileDefault {
+                    variant: H264ProfileDefaultVariant::High,
```

Comment:
I believe the default level should be Main.

Reply:
High was chosen for continuity with the existing default settings in some of the encoders.
Reviewing the diff, it is clear that most encoders default to High right now.

Reply:
@nowrep Any reason for this? Main should have less latency.

Reply:
Not sure how reliable this is but this says that high main and base have no difference in latency
https://ipvm.com/reports/h264-high-profile-tested
Given these are just cameras idk

Reply:
While higher profiles do increase complexity it seems it may not correlate to latency

Reply:
No opinion on this.

> Main should have less latency.

Did you test it?

Reply:
@nowrep No... it was just an assumption. also I don't use AMD.

Reply:
There's no latency difference between baseline and high on encoder side. I assume the goal of this PR is to reduce decoder latency?

Reply:
The PR goal is to generally reduce decoding requirements if the user so wishes. The reason may be due to decoder or encoder latency, or compatibility. In particular, pairing it with the software decoder fallback. A comment in the Moonlight source code suggests software decoders on hardware that is "supposed" to have hardware support are relatively limited on the profiles they support. Owing to that, these PRs are meant to work as a pair to act as a "pipeline of last resort".

If I can (somehow) figure out how to make ALVR work with my hardware properly without doing these sorts of things, I'll PR that as well, but I don't have high hopes at this point, owing to a pair of somewhat arcane problems I've been having on my end, both on the encoder and decoder side.

Reply:
If you want to fix software decoder compatibility then just use ffmpeg in the client instead of MediaCodec.

Reply:
While this is correct, in practice, AHardwareBuffers are involved, and it'd likely involve some refactoring, and there are APIs that expose these AHardwareBuffers. I don't have the necessary knowledge.

Reply:
I believe the encoding profile is still good to have.
@20kdc have you checked if the decoder can provide its own supported profiles? We could automate this as a future optimization

Reply:
As far as I'm aware decoder can't provide supported profiles outside of successful or failed initialization.
*Even if it could, decoder bugs make the output potentially untrustworthy.*
The *encoder* might be able to provide supported profiles to some extent, but that's more like an API-level thing, and it requires getting this information about support to the Dashboard, somehow.

Reply:
> As far as I'm aware decoder can't provide supported profiles outside of successful or failed initialization

That's too bad

> it requires getting this information about support to the Dashboard

No we don't. we can do the same thing we do for FPS, we just choose the closes one if unsupported by the client.